code/game/objects/effects/spawners/windowspawner.dm

```diff
@@ -8,33 +8,34 @@
 	anchored = 1 // No sliding out while you prime
 
 /obj/effect/spawner/window/Initialize()
-	..()
-	spawn(0)
-		var/turf/T = get_turf(src)
-		for(var/obj/structure/grille/G in get_turf(src))
-			// Complain noisily
-			log_runtime(EXCEPTION("Extra grille on turf: ([T.x],[T.y],[T.z])"), src)
-			qdel(G) //just in case mappers don't know what they are doing
+	. = ..()
+	var/turf/T = get_turf(src)
+	for(var/obj/structure/grille/G in get_turf(src))
+		// Complain noisily
+		log_runtime(EXCEPTION("Extra grille on turf: ([T.x],[T.y],[T.z])"), src)
+		qdel(G) //just in case mappers don't know what they are doing
 
-		if(!useFull)
-			for(var/cdir in cardinal)
-				for(var/obj/effect/spawner/window/WS in get_step(src,cdir))
-					cdir = null
-					break
-				if(!cdir)	continue
-				var/obj/structure/window/WI = new windowtospawn(get_turf(src))
-				WI.dir = cdir
-		else
-			var/obj/structure/window/W = new windowtospawn(get_turf(src))
-			W.dir = SOUTHWEST
+	if(!useFull)
+		for(var/cdir in cardinal)
+			for(var/obj/effect/spawner/window/WS in get_step(src,cdir))
+				cdir = null
+				break
+			if(!cdir)	continue
+			var/obj/structure/window/WI = new windowtospawn(get_turf(src))
+			WI.dir = cdir
+	else
+		var/obj/structure/window/W = new windowtospawn(get_turf(src))
+		W.dir = SOUTHWEST
 
-		if(useGrille)
-			new /obj/structure/grille(get_turf(src))
+	if(useGrille)
+		new /obj/structure/grille(get_turf(src))
 
-		src.air_update_turf(1) //atmos can pass otherwise
-
-		spawn(10)
-			qdel(src)
+	src.air_update_turf(1) //atmos can pass otherwise
+	// Give some time for nearby window spawners to initialize
+	spawn(10)
+		qdel(src)
+	// why is this line a no-op
+	// QDEL_IN(src, 10)
```

Comment:
For some reason when I used `QDEL_IN` here, the window spawners stuck around even after roundstart